File: V1-MVP版本/frontend/src/api/study.ts
/**
 * 学时档案 API 客户端。
 */
import { api } from '@/utils/request'

export interface StudyListItem {
  member_id: number
  member_name: string
  member_phone: string
  branch_name: string | null
  year: number
  total_hours: number
  activity_count: number
}

export interface StudyListResponse {
  year: number
  total_members: number
  items: StudyListItem[]
}

export interface StudyDetailItem {
  activity_id: number
  theme: string
  training_at: string
  location: string
  study_hours: number
  status: string
  attendance_status: string
}

export interface StudyDetailResponse {
  member_id: number
  member_name: string
  year: number
  total_hours: number
  activity_count: number
  items: StudyDetailItem[]
}

export const studyApi = {
  list(params: { year: number; community_id?: number; branch_id?: number }) {
    return api.get<StudyListResponse>('/study-hours', params)
  },
  me(year: number) {
    return api.get<StudyDetailResponse>('/study-hours/me', { year })
  },
  member(memberId: number, year: number) {
    return api.get<StudyDetailResponse>(`/study-hours/${memberId}`, { year })
  },
}

File: V1-MVP版本/backend/app/schemas/study.py
"""学时档案 schema。"""
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel, ConfigDict


class StudyHourSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    member_id: int
    year: int
    total_hours: Decimal
    activity_count: int
    updated_at: datetime

    member_name: str | None = None
    member_phone: str | None = None
    branch_name: str | None = None


class StudyHourListItem(BaseModel):
    member_id: int
    member_name: str
    member_phone: str
    branch_name: str | None = None
    year: int
    total_hours: Decimal
    activity_count: int


class StudyHourListResponse(BaseModel):
    year: int
    total_members: int
    items: list[StudyHourListItem]


class StudyDetailItem(BaseModel):
    """个人明细：每场活动的学时。"""
    activity_id: int
    theme: str
    training_at: datetime
    location: str
    study_hours: Decimal
    status: str
    attendance_status: str


class StudyDetailResponse(BaseModel):
    member_id: int
    member_name: str
    year: int
    total_hours: Decimal
    activity_count: int
    items: list[StudyDetailItem]
